### ISSUE TITLE
Is it possible to configure the message structure?

### ISSUE THREAD

**Issue by silverl** [ssplatt/slack-zenoss#4]
I've been using just the slack_zenoss.py as a command notifier for months, and liked the configurability of it.  I switched to the ZenPack/egg last week. There doesn't seem to be any way to configure which message elements are included, such as Component.

Am I just missing it? I'd like to have some fields to configure which of the available message components are passed along to Slack.

Thanks.

**Comment by silverl** [comment]
I took a look at the source code of slack_zenoss.py. It appears that while the script supports passing in the Component of the event, it does nothing with the value whatsoever.

**Comment by ssplatt** [comment]
basically, you'll need to modify slack_zenoss.py to fit whatever output you want.  It's not the most ideal situation but this is basically a hack to get any data in the first place. Also, the script only covers a small portion of the available expressions, so you can add more by hand if you want. For a full list of expressions, see http://community.zenoss.org/docs/DOC-12029

also, slack_zenoss.py isn't run by the .egg, it's been saved for historical reasons more or less.

**Comment by silverl** [comment]
There are two slack_zenoss.py in the repo.

```
./slack_zenoss.py
./ZenPacks/community/Slack/lib/slack_zenoss.py
```

They appear to be different. Are neither used?

If I wanted to grab one and make it my own, which do you recommend?

**Comment by ssplatt** [comment]
the one at root is the original. the other is a modification to work as a proper egg package.

**Comment by dstamen** [comment]
Is there a way to add some custom fields into the notification? we have a SaaS environment, so we dont have the ability to easily edit the zenpack and make changes.

**Comment by jregovic** [comment]
Yes, it really should have some more customization. My idea would be some kind of TALES evaluated fields to show allow a richer message.

**Comment by dstamen** [comment]
agree. it works good but would love to see component listed, we monitor alot of vsphere and switches, so most of the data we need is at the component level on alerts not just on the device.
